### PR TITLE
Support big strings

### PR DESCRIPTION
**Pull request by danobi**
Previously, all strings needed to be stored on the stack. Given that BPF
has a 512B stack limit, this put serious limits on strings. In practice,
it was limited to about 200B.

This PR raises the limit much higher to around 1024. The new limit
we hit is from the LLVM memset() builtin. It only supports up to 1K. To
go above this, we need to create our own memset() routine in BPF. This
is quite easy to do and will be attempted in a later commit.

The way big strings work is by assigning each str() call in the script a
unique ID. Then we create a percpu array map such that for each CPU,
there is a percpu entry that can accomodate each str() callsite. This
makes it so scripts can keep as many strings "on the stack" as they'd
like, with the tradeoff being the amount of memory we need to
preallocate.

As long as the kernel never allows BPF programs to nest (eg execute prog
A while prog A is already on the call stack), this approach is robust to
self-corruption.

This closes https://github.com/bpftrace/bpftrace/issues/305.

##### Checklist

- [ ] Language changes are updated in `man/adoc/bpftrace.adoc`
- [x] User-visible and non-trivial changes updated in `CHANGELOG.md`
- [x] The new behaviour is covered by tests

### CODE REVIEW

**Comment by danobi** [comment]
In another window:
```
$ /usr/bin/asdf/asdf/dsf/sdfddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
bash: /usr/bin/asdf/asdf/dsf/sdfddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd: No such file or directory
```

```
$ sudo BPFTRACE_MAX_STRLEN=1024 ./build/src/bpftrace -e 't:syscalls:sys_enter_execve { @[str(args.filename)] = count() }'
Attaching 1 probe...
^C

@[/usr/bin/asdf/asdf/dsf/sdfddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd]: 1
@[/home/dxu/.tmux/plugins/tmux-continuum/scripts/continuum_save.sh]: 2
@[/bin/sh]: 2
@[/usr/bin/cut]: 2
```

**Comment by danobi** [comment]
Related but not done: #2853

File: src/main.cpp [hunk 1]
```diff
@@ -269,25 +269,6 @@ static void parse_env(BPFtrace& bpftrace)
     config_setter.set(ConfigKeyInt::max_strlen, x);
   });
 
-  // in practice, the largest buffer I've seen fit into the BPF stack was 240
-  // bytes. I've set the bar lower, in case your program has a deeper stack than
-  // the one from my tests, in the hope that you'll get this instructive error
-  // instead of getting the BPF verifier's error.
-  uint64_t max_strlen = bpftrace.config_.get(ConfigKeyInt::max_strlen);
-  if (max_strlen > 200) {
```
Comment (jordalgo) [inline src/main.cpp:277]: why remove this check instead of updating it?

File: tests/runtime/call [hunk 2]
```diff
@@ -278,6 +278,8 @@ ENV BPFTRACE_CACHE_USER_SYMBOLS=PER_PROGRAM
 PROG uprobe:./testprogs/uprobe_symres_exited_process:test { print(ustack); exit(); }
 EXPECT_REGEX ^\s+test\+[0-9]+\s+test2\+[0-9]+\s+main\+[0-9]+
 AFTER ./testprogs/disable_aslr ./testprogs/uprobe_symres_exited_process
+# Required to skip function prologue
+REQUIRES_FEATURE dwarf
```
Comment (jordalgo) [inline tests/runtime/call:282]: driveby fixes?

**Comment by jordalgo** [comment]
This all looks great and surprisingly straightforward.

File: src/ast/passes/codegen_llvm.cpp [hunk 3]
```diff
@@ -719,37 +720,43 @@ void CodegenLLVM::visit(Call &call)
...
-                          expr_,
-                          arg0->type.GetAS(),
-                          call.loc);
-    b_.CreateLifetimeEnd(strlen);
+    b_.CreateProbeReadStr(
+        ctx_, buf, strlen, expr_, arg0->type.GetAS(), call.loc);
+    b_.CreateBr(lookup_merge_block);
+
+    // Think of this like an assert(). In practice, we cannot fail to lookup a
+    // percpu array map unless we have a coding error. Rather than have some
```
Comment (jordalgo) [inline src/ast/passes/codegen_llvm.cpp:750]: By "coding error" you mean we pass in a key that does not exist right?

File: src/main.cpp (hunk 1 above)
Comment (danobi) [reply src/main.cpp:277]: Yeah fair enough. I'll delete it when the memset() goes in.

File: tests/runtime/call (hunk 2 above)
Comment (danobi) [reply tests/runtime/call:282]: Oh, that change merged already. Lemme rebase onto it

File: src/ast/passes/codegen_llvm.cpp (hunk 3 above)
Comment (danobi) [reply src/ast/passes/codegen_llvm.cpp:750]: Yes

**Review by jordalgo — APPROVED** [review]
